usb_flash_drive_copy.py

```diff
@@ -0,0 +1,543 @@
+        return dict_videos
+
+
+@dataclass
+class UsbFlashDrive:
+    """Wrapper to usb flash drives providing mount and unmount function.
+
+    IMPORTANT: Requires an entry in /etc/fstab to give user permission to access
+    usb mount device.
+
+    `self.mount_point` must be the same as in /etc/fstab.
+
+    Raises:
+        UsbFlashDriveNotMountableError: If not able to mount usb flash drive.
+    """
+
+    mount_point: Path
+
+    def mount(self) -> None:
+        """Mount USB flash drive.
+
+        IMPORTANT: Requires an entry in /etc/fstab to give user permission to access
+        usb mount device.
+
+        `self.mount_point` must be the same as in /etc/fstab.
+        """
+        if self.mount_point.is_mount():
+            log.write("USB flash drive already mounted", log.LogLevel.WARNING)
+            return
+
+        self.mount_point.mkdir(parents=True, exist_ok=True)
+
+        return_code: int = subprocess.call(f"sudo mount {self.mount_point}", shell=True)
+        if return_code != 0:
+            raise UsbFlashDriveNotMountableError(
+                (f"Unable to mount USB flash drive to '{self.mount_point}'!",)
+            )
+        log.write("USB flash drive mounted.")
+
+    def unmount(self) -> None:
+        """Unmount USB flash drive."""
+        if not self.mount_point.is_mount():
+            log.write("USB flash drive already unmounted", log.LogLevel.WARNING)
+            return
+
+        return_code: int = subprocess.call(
+            f"sudo umount {self.mount_point}", shell=True
+        )
+        if return_code != 0:
+            raise UsbFlashDriveUnmountableError(
+                f"Unable to unmount USB flash drive from '{self.mount_point}'!"
+            )
+        log.write("USB flash drive unmounted.")
+
+
+class OTCameraUsbCopier(Observer):
+    def __init__(
+        self,
+        power_led: Led,
+        wifi_led: Led,
+        rec_led: Led,
+        src_dir: Path,
+        usb_flash_drive: UsbFlashDrive,
+    ) -> None:
+        """This classes' main purpose is to provide methods to copy and delete videos
+        with the help of a `CopyInformation` object.
+
+        Args:
+            power_led (Led): give user visual feedback to inform about the current
+            status of the OTCameraUsbCopier.
+            wifi_led (Led): give user visual feedback to inform about the current
+            status of the OTCameraUsbCopier.
+            rec_led (Led): give user visual feedback to inform about the current
+            status of the OTCameraUsbCopier.
+            src_dir (Path): directory where the video files to be copied are located at.
+            usb_flash_drive (UsbFlashdrive): The USB flash drive.
+        """
+        self.power_led = power_led
+        self.wifi_led = wifi_led
+        self.rec_led = rec_led
+        self.src_dir = src_dir
+        self.usb_flash_drive = usb_flash_drive
+
+    def update(self, subject: Subject) -> None:
+        if isinstance(subject, Button):
```

Comment:
To avoid type casts, the `is_active` attribute could be an attribute of the `update` method signature instead of the generic `Subject` class.

Reply:
Will do!